dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/config/DLCOracleAppConfig.scala

```diff
@@ -165,7 +165,13 @@ case class DLCOracleAppConfig(
                                           bip39PasswordOpt)
     val oracle = new DLCOracle(key)(this)
 
-    start().map(_ => oracle)
+    for {
+      _ <- start()
+      differentKeyDbs <- oracle.eventDAO.findDifferentPublicKey(
+        oracle.publicKey)
+      fixedDbs = differentKeyDbs.map(_.copy(pubkey = oracle.publicKey))
+      _ <- oracle.eventDAO.updateAll(fixedDbs)
```

Comment:
So we just update them with the new public key? Aren't there possibly announcements published with the old key?

Reply:
> So we just update them with the new public key? Aren't there possibly announcements published with the old key?

I don't think so, will be adding tests to make sure

Reply:
Added test, looks like we could have published an announcement on an old key :/. Not sure the best way to fix this

Reply:
I think we will have to support it then. Here is what I think the logic should be on next startup of a oracle 

1. Check if we have anything in the db
2. If we don't, use the new HDCoinType.DLC (I recommended above)
3. If we do, use `HDCoinType.Testnet` (we have no choice, i believe? Unless you want to create a new oracle)
4. Use the coin type to derive the correct `signingKey`. 

This should be well documented with links to issues that refer to what is going on. I don't think there has been a proper issue filed yet that details the issues here? #2748 just says there is a problem, with no detailed description of how the bug was introduced, why we didn't catch it until now, and long term implications of the bug (we now need to support two HDCoinType, or else have to explain forever why we are using testnet as our HDCoinType)

I do feel somewhat strongly about using a new HDCoinType to reduce amiguity of what is going on. As was our intention, we are trying to make the oracle agnostic, but we use network specific data structures in the code, which is confusing IMO.